### PR TITLE
NO-TICKET: Simplify linear_chain.

### PR DESCRIPTION
**Pull request by afck**
This fixes rustfmt and removes some unnecessary `Event` variants.

(Found in preparation of [NDRS-892](https://casperlabs.atlassian.net/browse/NDRS-892).)

### CODE REVIEW

File: node/src/components/linear_chain.rs [hunk 1]
```diff
@@ -270,10 +243,6 @@ impl<I> LinearChain<I> {
                 // TODO: disconnect from the sender.
                 continue;
             }
-            if known_signatures.proofs.contains_key(&fs.public_key) {
-                // Don't send finality signatures we already know of.
-                continue;
-            }
```
Comment (goral09) [inline node/src/components/linear_chain.rs:276]: Why did you remove this?

File: node/src/components/linear_chain.rs [hunk 2]
```diff
@@ -53,12 +53,6 @@ pub enum Event<I> {
         /// The deploys' execution results.
         execution_results: HashMap<DeployHash, ExecutionResult>,
     },
-    /// A continuation for `GetBlock` scenario.
-    GetBlockResult(BlockHash, Option<Box<Block>>, I),
-    /// A continuation for `BlockAtHeight` scenario.
-    GetBlockByHeightResult(u64, Option<Box<Block>>, I),
-    /// A continuation for `BlockAtHeightLocal` scenario.
-    GetBlockByHeightResultLocal(u64, Option<Box<Block>>, Responder<Option<Block>>),
```
Comment (goral09) [inline node/src/components/linear_chain.rs:61]: We never construct these events? So other components read directly from the storage? :grimacing:

Comment (afck) [reply node/src/components/linear_chain.rs:61]: No, the _requests_ are being made. These `*Result` events were just used internally, sent from `LinearChain` to itself, instead of doing everything in an `async` block.

File: node/src/components/linear_chain.rs (hunk 1 above)
Comment (afck) [reply node/src/components/linear_chain.rs:276]: It's duplicated: It's already in line 236.

**Review by goral09 — APPROVED** [review]
